### PR TITLE
feat: add Close Approach module

### PR DESCRIPTION
**Pull request by phc1990**
Adds a Close Approach module:
- Close Approach
- Close Approach Generator

Example:

```python

generator: Generator = Generator(
    reference_trajectory,
    step_size,
    tolerance,
)

close_approaches: list[CloseApproach] = []

for trajectory in other_objects_trajectories:
    close_approaches.extend(
        generator.compute_close_approaches(
            trajectory,
            search_interval,
        )

for close_approach in close_approaches:
    print(f"{close_approach.get_instant()}, {close_approach.get_miss_distance().in_meters()}"
```     

## Summary by CodeRabbit

* **New Features**
  * Added close-approach analysis: CloseApproach objects with miss-distance and relative-state access, component breakdowns in arbitrary and local orbital frames, and event detection over time windows with configurable step and tolerance.
  * Made functionality available in both C++ and Python, including a dedicated Python close_approach submodule.

* **Tests**
  * Added comprehensive unit and integration tests covering CloseApproach and Generator behavior, computations, and edge cases.

### CODE REVIEW

File: bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Conjunction/Generator.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+        .def(
+            init<const Trajectory&, const Duration&, const Duration&>(),
+            R"doc(
+                Constructor.
+
+                Args:
+                    reference_trajectory (Trajectory): The reference trajectory for which to find close approaches (Object 1).
+                    search_step_size (Duration): The step size to use during the close approach search. Defaults to Duration.seconds(15.0).
+                    search_tolerance (Duration): The tolerance to use during the close approach search. Defaults to Duration.milliseconds(1.0).
```
Comment (vishwa2710) [inline bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Conjunction/Generator.cpp]:
```suggestion
                    step (Duration): The step size to use during the close approach search. Defaults to Duration.seconds(15.0).
                    tolerance (Duration): The tolerance to use during the close approach search. Defaults to Duration.milliseconds(1.0).
```
To be consistent with the access generator.

File: bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Conjunction/Generator.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            "set_search_tolerance",
+            &Generator::setSearchTolerance,
+            R"doc(
+                Set the search tolerance.
+
+                Args:
+                    search_tolerance (Duration): The search tolerance.
+            )doc",
+            arg("search_tolerance")
+        )
```
Comment (vishwa2710) [inline bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Conjunction/Generator.cpp]:
```suggestion
```
I'd remove the setters as we typically do not provide setters for trivial types like this, you can just create a new generator.

File: include/OpenSpaceToolkit/Astrodynamics/Conjunction/Generator/Generator.hpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    /// @brief Get the search step size
+    ///
+    /// @code{.cpp}
+    ///              Generator generator = { ... } ;
+    ///              Duration stepSize = generator.getSearchStepSize() ;
+    /// @endcode
+    ///
+    /// @return The search step size
+    Duration getSearchStepSize() const;
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/Astrodynamics/Conjunction/Generator/Generator.hpp]: rename to step to align with generator.
```suggestion
    Duration getStep() const;
```

File: src/OpenSpaceToolkit/Astrodynamics/Conjunction/Generator/Generator.cpp [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        // Create close approach at the start of the interval (when objects transition from approaching to moving
+        // apart)
+        const Instant closeApproachInstant = interval.getStart();
+        const State object1State = referenceTrajectory_.getStateAt(closeApproachInstant);
+        const State object2State = anOtherObjectTrajectory.getStateAt(closeApproachInstant);
+
+        closeApproaches.add(CloseApproach(object1State, object2State));
+    }
+
+    // Sort close approaches by instant
```
Comment (vishwa2710) [inline src/OpenSpaceToolkit/Astrodynamics/Conjunction/Generator/Generator.cpp]:
```suggestion
```

**Review by vishwa2710 — APPROVED** [review]
Mostly nits, great stuff! Thanks for doing this.
